### PR TITLE
Access RunIndex from RunForOutput

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

This adds the index() method to RunForOutput. This is needed to allow use of run caches in OutputModules

#### PR validation:

The code compiles and does allow use of RunCache from a global OutputModule.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Framework/interface/RunForOutput.h [hunk 1]
```diff
@@ -57,6 +58,10 @@ namespace edm {
     Timestamp const& endTime() const { return aux_.endTime(); }
     MergeableRunProductMetadata const* mergeableRunProductMetadata() const { return mergeableRunProductMetadata_; }
 
+    /**\return Reusable index which can be used to separate data for different simultaneous LuminosityBlocks.
```
Comment (makortel) [inline FWCore/Framework/interface/RunForOutput.h:61]: I'm confused, how does `RunIndex` help to differentiate `LuminosityBlock`s?

Comment (Dr15Jones) [reply FWCore/Framework/interface/RunForOutput.h:61]: It doesn't, that is a cut-n-paste error :)

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**